core/src/main/java/org/apache/calcite/sql/SqlDeclareHandler.java

```diff
@@ -0,0 +1,106 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to you under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.calcite.sql;
+
+import org.apache.calcite.sql.parser.SqlParserPos;
+import org.apache.calcite.util.ImmutableNullableList;
+
+import java.util.List;
+import java.util.Objects;
+
+/**
+ * Parse tree for {@code SqlDeclareHandler} call.
+ */
+public class SqlDeclareHandler extends SqlCall {
+  private static final SqlSpecialOperator OPERATOR =
+      new SqlSpecialOperator("DECLARE_HANDLER", SqlKind.DECLARE_HANDLER);
+
+  public final HandlerType handlerType;
+  public final SqlIdentifier conditionName;
+  public final SqlNodeList parameters;
+  public final SqlNode handlerStatement;
+
+  /**
+   * Creates a {@code SqlDeclareHandler}.
+   *
+   * @param pos Parser position, must not be null
+   * @param handlerType The type of handler being declared, must not be null
+   * @param conditionName Name of condition before CONDITION keyword
+   * @param parameters List of SQLSTATE values, or
+   *                   {@link SqlDeclareHandlerCondition} objects and
+   *                   identifiers, must not be null
+   * @param handlerStatement Handler action statement
+   */
+  public SqlDeclareHandler(SqlParserPos pos, HandlerType handlerType,
+      SqlIdentifier conditionName, SqlNodeList parameters,
+      SqlNode handlerStatement) {
+    super(pos);
+    this.handlerType = Objects.requireNonNull(handlerType);
+    this.conditionName = conditionName;
+    this.parameters = Objects.requireNonNull(parameters);
+    this.handlerStatement = handlerStatement;
+  }
+
+  @Override public SqlOperator getOperator() {
+    return OPERATOR;
+  }
+
+  @Override public List<SqlNode> getOperandList() {
+    return ImmutableNullableList.of(conditionName, parameters,
+        handlerStatement);
+  }
+
+  @Override public void unparse(SqlWriter writer, int leftPrec, int rightPrec) {
+    writer.keyword("DECLARE");
+    switch (handlerType) {
+    case CONTINUE:
```

Comment:
nit: indent case statements

Reply:
Drive-by comment: the style used in this code base is to not indent the case beyond the switch.